include/rocksdb/wide_columns.h

```diff
@@ -218,4 +224,49 @@ inline bool operator!=(const PinnableWideColumns& lhs,
   return !(lhs == rhs);
 }
 
+inline PinnableWideColumns::PinnableWideColumns(const PinnableWideColumns& p)
+    : columns_(p.columns_) {
+  CopyValue(p.value_);
+}
+
+class GroupedPinnableWideColumns {
+ public:
+  const size_t& num_column_families() const { return num_column_families_; }
+  const std::vector<Status>& statuses() const { return statuses_; }
+  const std::vector<PinnableWideColumns>& columns() const { return columns_; }
+  PinnableWideColumns* columns_ref(const size_t& column_family_index) {
+    assert(column_family_index < num_column_families_);
+    return &columns_[column_family_index];
+  }
+
+  explicit GroupedPinnableWideColumns(const size_t& num_column_families)
+      : num_column_families_(num_column_families),
+        statuses_(num_column_families),
+        columns_(num_column_families) {}
+
+  void SetStatus(size_t column_family_index, Status&& status);
+  void SetStatusForAllColumnFamilies(Status status);
+
+  void Reset();
+
+ private:
+  const size_t num_column_families_;
+  std::vector<Status> statuses_;
+  std::vector<PinnableWideColumns> columns_;
+};
+
+inline void GroupedPinnableWideColumns::SetStatus(size_t column_family_index,
+                                                  Status&& status) {
+  statuses_[column_family_index] = status;
```

Comment:
status copy by operator= should be cheap, but still I'm wondering if there's better way to set status for each CF get result.

Reply:
Yep, copying `Status` is fine; it's a lightweight operation